api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java

```diff
@@ -23,109 +23,235 @@
 import jakarta.data.mock.entity._Book;
 
 import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.Set;
+
 import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
 
 
 class BasicRestrictionRecordTest {
 
+
     @Test
+    @DisplayName("should create an EqualTo restriction by default when using 'equalTo' without negation")
     void shouldCreateBasicRestrictionWithDefaultNegation() {
-        @SuppressWarnings("unchecked")
```

Comment:
Getting rid of `SuppressWarnings` causes complaints like the following to show up in the build output. I would say keep it and don't remove it.  Also, new tests that you added ought to include it where needed.
```
[...]/data/api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java: [...]/data/api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java uses unchecked or unsafe operations.
[...]/data/api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java: Recompile with -Xlint:unchecked for details.
```